src/david2.f90

```diff
@@ -283,7 +283,7 @@ subroutine solver_sdavidson(n,crite,Hp,C,e,fail,pr)
    e    = 0
 
    do i=1,n
-      adiag(i)=HP(i*(i-1)/2)
+      adiag(i)=HP(i*(i+1)/2)
```

Comment:
Diagonal element is given by `i*(i-1)/2+i` == `i*(i+1)/2`. Fixes out-of-bound access.